### PR TITLE
Persist time point for solar light direction.

### PR DESCRIPTION
**Pull request by pmconne**
Design Review team is implementing a UI to control the solar direction based on the date/time and location. `DisplayStyleState.setSunTime` can compute and persist the direction from the time but the time itself is not persisted anywhere.

Added to SolarLightProps an optional `timePoint` to store the time from which the direction was derived.
Moved `setSunTime` to `DisplayStyleSettings` so it can also be used by backend too.
Added APIs to get and clear the time point.
Adjusted `LightSettings.clone` to clear the time point if it was previously defined and the caller supplied a new direction but not a new time point - unless the new direction matches the previous direction.

### CODE REVIEW

File: core/common/src/DisplayStyleSettings.ts [hunk 1]
```diff
@@ -1364,6 +1367,54 @@ export class DisplayStyle3dSettings extends DisplayStyleSettings {
     this._json3d.lights = lights.toJSON();
   }
 
+  /** Adjust the solar light direction based on a date and time at a geographic location.
+   * This replaces `this.lights` with a copy that records the time point and the computed direction.
+   * @param timePoint The time in UNIX milliseconds.
+   * @param location The geographic location, or an iModel from which the geolocation is to be obtained.
+   * @see [[sunTime]] to get the current sun time.
+   * @see [[clearSunTime]] to clear the time point.
+   * @note If the iModel is not geolocated, a location in Exton, Pennsylvania will be used to compute the light direction.
```
Comment (kabentley) [inline core/common/src/DisplayStyleSettings.ts:1376]: maybe: "if `location` is an iModel, but it is not geolocated..."

File: core/common/src/DisplayStyleSettings.ts [hunk 2]
```diff
@@ -1364,6 +1367,54 @@ export class DisplayStyle3dSettings extends DisplayStyleSettings {
...
+   * @see [[sunTime]] to get the current sun time.
+   * @see [[clearSunTime]] to clear the time point.
+   * @note If the iModel is not geolocated, a location in Exton, Pennsylvania will be used to compute the light direction.
+   */
+  public setSunTime(timePoint: number, location: IModel | Cartographic): void {
+    let cartoCenter;
+    if (location instanceof IModel) {
+      if (location.isGeoLocated) {
+        const projectExtents = location.projectExtents;
+        const projectCenter = Point3d.createAdd2Scaled(projectExtents.low, .5, projectExtents.high, .5);
```
Comment (kabentley) [inline core/common/src/DisplayStyleSettings.ts:1383]: @RBBentley  wouldn't it be better to use the ecef origin?

File: core/common/src/LightSettings.ts [hunk 3]
```diff
@@ -24,13 +24,17 @@ function extractIntensity(value: number | undefined, defaultValue: number) {
 export interface SolarLightProps {
   /** Intensity of the light, typically in [0..1] but can range up to 5. Default: 1.0 */
   intensity?: number;
-  /** Direction of the light in world coordinates. Default: (-1, -1, -1) */
+  /** Direction of the light in world coordinates. Default: (0.272166, 0.680414, 0.680414) */
```
Comment (kabentley) [inline core/common/src/LightSettings.ts:27]: what do these numbers mean?

File: core/common/src/DisplayStyleSettings.ts (hunk 2 above)
Comment (pmconne) [reply core/common/src/DisplayStyleSettings.ts:1383]: Ask @RBBentley - I just moved his code.

File: core/common/src/LightSettings.ts (hunk 3 above)
Comment (pmconne) [reply core/common/src/LightSettings.ts:27]: This was our ancient default directional light direction from DgnClientSdk days, before we made the lighting configurable.
I can add an explanation.

File: core/common/src/DisplayStyleSettings.ts (hunk 2 above)
Comment (RBBentley) [reply core/common/src/DisplayStyleSettings.ts:1383]: No - this is fine.   We don't really store a seperate ECEF origin so project center is best we can do.

File: core/common/src/LightSettings.ts (hunk 3 above)
Comment (RBBentley) [reply core/common/src/LightSettings.ts:27]: This is just some random light direction - not sure where it came from originally but it provides a decentt shadows in absence of a real direction.

**Review by RBBentley — APPROVED** [review]

File: core/common/src/DisplayStyleSettings.ts (hunk 2 above)
Comment (kabentley) [reply core/common/src/DisplayStyleSettings.ts:1383]: we do store the ecef center - that's what the ecef transform means. If the iModel is projected, we calculate the ecef center using this algorithm (though i thinlk we adjust z=0). If the iModel is not projected, the ecef origin comes from the application.

File: core/common/src/DisplayStyleSettings.ts (hunk 1 above)
Comment (pmconne) [reply core/common/src/DisplayStyleSettings.ts]: tweaked wording.

File: core/common/src/DisplayStyleSettings.ts (hunk 2 above)
Comment (pmconne) [reply core/common/src/DisplayStyleSettings.ts]: changed to use ecef location.

File: core/common/src/LightSettings.ts (hunk 3 above)
Comment (pmconne) [reply core/common/src/LightSettings.ts]: translated numbers to English.

**Comment by pmconne** [comment]
@kabentley do you approve?

**Review by kabentley — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Comment by pmconne** [comment]
@Mergifyio backport release/2.12.x

**Pull request merged**